Source/JavaScriptCore/assembler/AssemblerCommon.h

```diff
@@ -376,31 +378,51 @@ static ALWAYS_INLINE void* memcpyAtomicIfPossible(void* dst, const void* src, si
     case 8:
         WTF::atomicStore(std::bit_cast<uint64_t*>(dst), *std::bit_cast<const uint64_t*>(src), std::memory_order_relaxed);
         return dst;
+    case 16: {
+        struct v128_t {
+            uint64_t x[2];
+        };
+        WTF::atomicStore(std::bit_cast<v128_t*>(dst), *std::bit_cast<const v128_t*>(src), std::memory_order_relaxed);
+        return dst;
+    }
     default:
         break;
     }
-#endif
+    RELEASE_ASSERT_NOT_REACHED();
+    return nullptr;
+}
+
+static ALWAYS_INLINE void* memcpyNonAtomic(void* dst, const void* src, size_t n)
+{
+    // We should expect these instructions to be torn, so let's verify that.
+    if (Options::fuzzAtomicJITMemcpy()) [[unlikely]] {
+        auto* d = reinterpret_cast<uint8_t*>(dst);
+        auto* s = reinterpret_cast<const uint8_t*>(src);
+        for (size_t i = 0; i < n; ++i, ++s, ++d) {
+            *d = *s;
+            WTF::storeLoadFence();
+        }
+    }
     return memcpy(dst, src, n);
 }
 
-static void* performJITMemcpy(void* dst, const void* src, size_t n);
+static ALWAYS_INLINE void* memcpyAtomicIfPossible(void* dst, const void* src, size_t n)
+{
+    if (n == 1 || n == 2 || n == 4 || n == 8)
```

Comment:
Should we have `n ==16` too?

Reply:
In the follow-up, I plan to. I just wanted to make sure this patch didn't change behaviour at first.